plugins/widget/plugin.js

```diff
@@ -3078,28 +3078,46 @@
 		var changed = 0,
 			classes = getStyleClasses( style ),
 			updatedClasses = widget.data.classes || {},
-			cl;
+			cl,
+			styles = style.getDefinition().styles,
+			attributes = style.getDefinition().attributes;
 
 		// Ee... Something is wrong with this style.
-		if ( !classes )
+		if ( !classes && !styles && !attributes )
 			return;
 
-		// Clone, because we need to break reference.
-		updatedClasses = CKEDITOR.tools.clone( updatedClasses );
+		if ( classes ) {
+			// Clone, because we need to break reference.
+			updatedClasses = CKEDITOR.tools.clone( updatedClasses );
 
-		while ( ( cl = classes.pop() ) ) {
-			if ( apply ) {
-				if ( !updatedClasses[ cl ] )
-					changed = updatedClasses[ cl ] = 1;
-			} else {
-				if ( updatedClasses[ cl ] ) {
-					delete updatedClasses[ cl ];
-					changed = 1;
+			while ( ( cl = classes.pop() ) ) {
+				if ( apply ) {
+					if ( !updatedClasses[ cl ] )
+						changed = updatedClasses[ cl ] = 1;
+				} else {
+					if ( updatedClasses[ cl ] ) {
+						delete updatedClasses[ cl ];
+						changed = 1;
+					}
+				}
+			}
+			if ( changed )
+				widget.setData( 'classes', updatedClasses );
+		}
+
+		function toggleStyleAttribute ( element, property, StyleOrAttr, apply ) {
+			apply = apply ? 'set' : 'remove';
+			StyleOrAttr = StyleOrAttr ? 'Style' : 'Attribute';
+			StyleOrAttr = apply + StyleOrAttr;
+
+			for ( var key in property ) {
```

Comment:
Instead of iterating you may use `apply/removeStyles` and `apply/removeAttributes`.

Reply:
1. There is no method `removeStyles` only `removeStyle`.
2. Method `setAttributes` takes `object` containing pairs as attribute, while `removeAttributes` takes `array` as argument.

Applying all that will make our logic more complicated than it is right now so I will stay with current approach.